File: Top_Interview_150/Array_String/cpp/189.cpp
#include <vector>

using namespace std;

class Solution
{
public:
    void rotate(vector<int> &nums, int k)
    {
        int n = nums.size();
        k %= n;
        vector<int> offset(n, 0);
        for (int i = 0; i < n - k; ++i)
        {
            offset[i] = nums[i];
        }

        int count = 0;
        int i = n - k;
        while (count < k)
        {
            nums[count] = nums[i];
            count++;
            i++;
        }

        for (int i = k; i < n; ++i)
        {
            nums[i] = offset[i - k];
        }
    }
};

File: Top_Interview_150/Array_String/python/189.py
from typing import List


class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        n = len(nums)
        k = k % n
        offset = nums[:n - k]

        count = 0
        i = n - k
        while count < k:
            nums[count] = nums[i]
            count += 1
            i += 1

        for i in range(k, n):
            nums[i] = offset[i - k]

        print(nums)


solution = Solution()
print(solution.rotate(nums=[-1], k=2))
